pastas_plugins/pest/solver.py

```diff
@@ -1274,3 +1300,323 @@ def solve() -> None:
             "PestSenSolver does not have a solve method. Run the sensitivity"
             "analysis using the `start` method."
         )
+
+
+class RandomizedMaximumLikelihoodSolver(BaseSolver):
+    _name = "RandomizedMaximumLikelihoodSolver"
+
+    def __init__(
+        self,
+        num_reals: int,
+        jacobian_method: Literal["2-point", "3-point", "empirical"] = "3-point",
+        noptmax: int | None = None,
+        seed: int | None = pyemu.en.SEED,
+        add_base: bool = True,
+        num_workers: int | None = None,
+        pcov: pd.DataFrame | None = None,
+        nfev: int | None = None,
+        **kwargs,
+    ):
+        super().__init__(pcov=pcov, nfev=nfev, **kwargs)
+        self.num_reals = num_reals
+        self.jacobian_method = jacobian_method
+        if noptmax is None and jacobian_method == "empirical":
+            logger.error(
+                "noptmax must be specified when using 'empirical' jacobian method."
+            )
+        self.noptmax = noptmax
+        self.seed = seed
+        self.add_base = add_base
+        self.num_workers = cpu_count() if num_workers is None else num_workers
+        self.parameter_ensemble: pd.DataFrame | None = None
+        self.observation_noise: pd.DataFrame | None = None
+        self.simulation_ensemble: pd.DataFrame | None = None
+
+    def __repr__(self) -> str:
+        return f"RandomizedMaximumLikelihoodSolver(num_reals={self.num_reals}, jacobian_method={self.jacobian_method})"
+
+    def to_dict(self) -> dict:
+        data = {
+            "class": self._name,
+            "num_reals": self.num_reals,
+            "jacobian_method": self.jacobian_method,
+            "seed": self.seed,
+        }
+        return data
+
+    def initialize(
+        self,
+        standard_deviation: float = 0.0,
+        correlation_coefficient: float = 0.0,
+        par_sigma_range: float = 4.0,
+        method: Literal["norm", "truncnorm", "uniform"] = "norm",
+    ) -> None:
+        logger.debug("Initialize: RML solver")
+        logger.debug("Initialize: Creating observation noise")
+        observations = self.ml.observations()
+        observation_noise = pd.DataFrame(
+            PestIesSolver.generate_observation_noise(
```

Comment:
This might be a separate function? Perhaps a separate issue, but having a synthetic time series generator and noise generator might be useful for multiple applications within pastas-world.